lib/cartopy/tests/test_img_tiles.py

```diff
@@ -285,3 +288,91 @@ def test_ordnance_survey_get_image():
 
     # The extent is the same though
     assert extent1 == extent2
+
+
+@pytest.mark.network
+@pytest.mark.parametrize('cache_dir', ["tmpdir", True, False])
+def test_cache(cache_dir, tmpdir):
+    if cache_dir == "tmpdir":
+        tmpdir_str = tmpdir.strpath
+    else:
+        tmpdir_str = cache_dir
+
+    if cache_dir is True:
+        config["cache_dir"] = tmpdir.strpath
+
+    # Fetch tiles and save them in the cache
+    with warnings.catch_warnings(record=True) as w:
+        gt = cimgt.GoogleTiles(cache=tmpdir_str)
+    gt._image_url = types.MethodType(GOOGLE_IMAGE_URL_REPLACEMENT, gt)
+
+    ll_target_domain = sgeom.box(-10, 50, 10, 60)
+    multi_poly = gt.crs.project_geometry(ll_target_domain, ccrs.PlateCarree())
+    target_domain = multi_poly.geoms[0]
+
+    img_init, _, _ = gt.image_for_domain(target_domain, 6)
+
+    # Do not check the result if the cache is disabled
+    if cache_dir is False:
+        assert gt.cache_path is None
+        return
+
+    # Check that the warning is properly raised (only when cache is True)
+    if cache_dir is True:
+        assert len(w) == 1
+    else:
+        assert len(w) == 0
+
+    # Define expected results
+    x_y_z_f_h = [
+        (30, 18, 6, '30_18_6.npy', '545db25f1aa348ad85e1f437fd0db0d9'),
```

Comment:
What's the thinking on checking the hashes? Won't this test break if Google updates the images?

Reply:
I used hashes to check that the proper tiles are downloaded and that the cache is properly used when the same tiles are used a second time (this is why is update the cache images before the second call, so I can check that they are not downloaded again).
Yes it will break if these tiles change. But as far as I know, the tiles returned by the URL stored in `GOOGLE_IMAGE_URL_REPLACEMENT` should never change (it is already used in some other tests).

Reply:
Fair enough.